docs/adrs/00009-new-field-license.md

```diff
@@ -0,0 +1,160 @@
+
+- If the license is an ID, the value is saved directly in the text field and spdx_licenses.
+- If the license is a name: The value is saved directly in the text field, and both spdx_licenses and spdx_license_exceptions remain empty.
+- If the license is an expression: Trustify parses it into standard SPDX license IDs and exceptions, stores them in spdx_licenses and spdx_license_exceptions, and also preserves the original expression in the text field.
+
+## The Current Issue
+
+- In CycloneDX, when the license type is 'name', Trustify only stores it in a text field, and we cannot determine whether this field represents just a name.
+- In SPDX, when a license expression contains a custom license ID, Trustify also saves it only in a text field, and filters out the custom license ID during the process of parsing it into standard SPDX license IDs and exceptions.
+- In SPDX, when a license expression includes a custom license ID, the custom license ID itself is meaningless and only has significance within the current SBOM, so it cannot be used for statistics or searches.
+
+## Decision
+
+Add a new field named `custom_license_refs` to the license table. Its type is a list, which includes licenses of type name appearing in CycloneDX, as well as license expressions that contain custom licenses in SPDX.
+
+```sql
+ALTER TABLE license ADD COLUMN custom_license_refs text[];
+```
+
+### CycloneDX
+
+Licenses of type 'name':
+
+```json
+{
+  "bom-ref": "pkg:rpm/centos/audit-libs@3.0-0.17.20191104git1c2f876.el8?arch=x86_64&upstream=audit-3.0-0.17.20191104git1c2f876.el8.src.rpm&distro=centos-8&package-id=05587266ec4157c9",
+  "type": "library",
+  "publisher": "CentOS",
+  "name": "audit-libs",
+  "version": "3.0-0.17.20191104git1c2f876.el8",
+  "licenses": [
+    {
+      "license": {
+        "name": "LGPLv2+"
+      }
+    }
+  ],
+  "cpe": "cpe:2.3:a:audit-libs:audit-libs:3.0-0.17.20191104git1c2f876.el8:*:*:*:*:*:*:*",
+  "purl": "pkg:rpm/centos/audit-libs@3.0-0.17.20191104git1c2f876.el8?arch=x86_64&upstream=audit-3.0-0.17.20191104git1c2f876.el8.src.rpm&distro=centos-8",
+  "properties": [
+    {
+      "name": "syft:package:foundBy",
+      "value": "rpm-db-cataloger"
+    }
+  ]
+}
+```
+
+`custom_license_refs` should be a JSONB mapping all custom license identifiers to their descriptions or SPDX expressions.
+
+Example:
+
+```json
+{
+  "LGPLv2+": "LGPLv2+"
+}
+```
+
+### SPDX
+
+License expression:
+
+```json
+{
+  "SPDXID": "SPDXRef-2a02a923-8a04-489d-9cbc-80f2d23de5ea",
+  "copyrightText": "NOASSERTION",
+  "downloadLocation": "https://access.redhat.com/downloads/content/package-browser",
+  "externalRefs": [
+    {
+      "referenceCategory": "PACKAGE_MANAGER",
+      "referenceLocator": "pkg:rpm/redhat/foreman-bootloaders-redhat@202102220000-1.el8sat?arch=src",
+      "referenceType": "purl"
+    }
+  ],
+  "filesAnalyzed": false,
+  "licenseConcluded": "NOASSERTION",
+  "licenseDeclared": "LicenseRef-2 AND LicenseRef-11 AND LicenseRef-BSD",
```

Comment:
How would that work with SPDX licenses IDs (non-custom)?

Reply:
It will parse such license expressions and extract the standard license IDs, storing them in spdx_licenses and spdx_license_exceptions.